### PR TITLE
Focus last active element on close

### LINKED ISSUE #23: Set focus on open, and Restore focus on close.
Focus is not restored when closing the overlay, last active element in the document should be focused

### PR DESCRIPTION
**Pull request by samiheikki**
Fixes #23

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-overlay/132)

### CODE REVIEW

File: src/vaadin-overlay.html [hunk 1]
```diff
@@ -527,6 +527,7 @@
         }
 
         if (opened) {
+          this.__previouslyFocusedElement = document.activeElement;
```
Comment (tomivirkki) [inline src/vaadin-overlay.html]: `document.activeElement` wouldn't work inside a shadow root. Need to use `getRootNode().activeElement`. Add a test for this case also.

Comment (web-padawan) [reply src/vaadin-overlay.html]: Done. Also, put the implementation behind the flag to make it opt-in.

File: src/vaadin-overlay.html [hunk 2]
```diff
@@ -604,6 +614,20 @@
...
+            if (this.restoreFocusOnClose && this.__restoreFocusNode) {
+              // If the activeElement is `<body>` or inside the overlay,
+              // we are allowed to restore the focus. In all the other
+              // cases focus might have been moved elsewhere by another
+              // component or by the user interaction (e.g. click on a
+              // button outside the overlay).
+              const activeElement = this._getActiveElement();
+
+              if (activeElement === document.body || this._deepContains(activeElement)) {
+                this.__restoreFocusNode.focus();
```
Comment (tomivirkki) [inline src/vaadin-overlay.html:610]: Works inconsistently on IE11:
- Closing with esc: the focus restores properly
- Closing with outside click: the focus is lost

**Review by tomivirkki — CHANGES_REQUESTED** [review]

File: src/vaadin-overlay.html (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-overlay.html:610]: Done. Added the check for `document.documentElement` where the focus is moving in that case.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**